crates/core/src/fs/backup.rs

```diff
@@ -11,7 +11,7 @@ use zip::{write::FileOptions, CompressionMethod, ZipWriter};
 
 /// A trait defining a way to back things up to the fs
 pub trait Backup {
-    fn backup(&self) -> Result<()>;
+    fn backup(&self, level: Option<i32>) -> Result<()>;
```

Comment:
Should `level` be a `ZstdBackup` field instead of requiring it in the trait method, since not all trait impls use `level`, like `ZipBackup`?

Reply:
Yes, good point. I don't know why I didn't think about that. It has no purpose in a generic trait like that. It has been removed now.